mmdet/apis/train.py

```diff
@@ -138,8 +139,14 @@ def train_detector(model,
     else:
         optimizer_config = cfg.optimizer_config
 
+    if cfg['lr_config']['policy'] == 'CosineAnealingUntilEpoch':
```

Comment:
I guess it can work without this `if..else` block. Have a look at mmcv.runner register_training_hooks implementation and rename your hook accordingly